### ISSUE TITLE
Clarification / potential bug: grayscale → RGB conversion does not replicate non-zero channel across 3 channels

### ISSUE THREAD

**Issue by vadori** [MouseLand/cellpose#1404]
Hi Cellpose-SAM authors,

I’m working with single-channel microscopy images and noticed a possible inconsistency between the paper description and the current preprocessing implementation for grayscale → RGB conversion.

In the manuscript, you state that: “For single-channel images, grayscale conversion was done by replicating the non-zero channel across all three channels.”

However, the current code path for 2D inputs appears to populate only one channel rather than replicating across all three.

```
if ndim == 2:
    # grayscale image, make 3 channels
    x_out = np.zeros((x.shape[0], x.shape[1], 3), dtype=x.dtype)
    x_out[..., 0] = x
    x = x_out
    del x_out
```

This produces:

Channel 0 → grayscale image

Channel 1 → zeros

Channel 2 → zeros

i.e.

RGB = [x, 0, 0]

So, could you please clarify if Cellpose-SAM was trained according to the paper or the github code?

**Comment by mrariden** [comment]
Hi @vadori, the code you're referencing is for image loading during evaluation. Training was done as described in the paper. We've not posted the training code yet but it will be released with publication of the paper

**Comment by vadori** [comment]
Thanks for the clarification @mrariden. The code is also used during training in the repository, not just for evaluation. Not the exact snippet I posted earlier, but the function pasted below — specifically the line:

`td = np.stack((td, 0*td, 0*td), axis=0)`

which has the same effect (i.e., populating only one channel). 

```
def _reshape_norm(data, channel_axis=None, normalize_params={"normalize": False}):
    """
    Reshapes and normalizes the input data.

    Args:
        data (list): List of input data, with channels axis first or last.
        normalize_params (dict, optional): Dictionary of normalization parameters. Defaults to {"normalize": False}.

    Returns:
        list: List of reshaped and normalized data.
    """
    if (np.array([td.ndim!=3 for td in data]).sum() > 0 or
        np.array([td.shape[0]!=3 for td in data]).sum() > 0):
        data_new = []
        for td in data:
            if td.ndim == 3:
                channel_axis0 = channel_axis if channel_axis is not None else np.array(td.shape).argmin()
                # put channel axis first 
                td = np.moveaxis(td, channel_axis0, 0)
                td = td[:3] # keep at most 3 channels
            if td.ndim == 2 or (td.ndim == 3 and td.shape[0] == 1):
                td = np.stack((td, 0*td, 0*td), axis=0) # ! should be td = np.stack((td, td, td), axis=0)
            elif td.ndim == 3 and td.shape[0] < 3:
                td = np.concatenate((td, 0*td[:1]), axis=0)
            data_new.append(td)
        data = data_new
    if normalize_params["normalize"]:
        data = [
            normalize_img(td, normalize=normalize_params, axis=0)
            for td in data
        ]
    return data

```

**Comment by vadori** [comment]
Closing.
